### ISSUE TITLE
Password must meet complexity requirements

### ISSUE THREAD

**Issue by DirectXGuy** [unosquare/passcore#278]
**PassCore Server**
[Windows 10 server 16]
[Active Directory]
Settings file
{
  "Logging": {
    "IncludeScopes": false,
    "LogLevel": {
      "Default": "Debug",
      "System": "Information",
      "Microsoft": "Information"
    }
  },
  "WebSettings": {
    "EnableHttpsRedirect": true
  },
  "AppSettings": {
    // The following options for AD Provider (remove if you don't use this Provider)
    "UseAutomaticContext": true, // Set true to allow PassCore to reset password using the same credentials, or false if you will fill the credentials below
    "RestrictedADGroups": [
      "Administrators",
      "Domain Admins",
      "Enterprise Admins"
    ], // Set the AD groups to restrict the use of PassCore
    "AllowedADGroups": [], // Set the AD Groups to allow PassCore, if the array is empty all the groups no-restricted above are allowed
    "IdTypeForUser": "UPN", // Possible values are "DN", "GUID", "Name", "SAM", "SID" and "UPN" (Default UPN)
    "UpdateLastPassword": false, // Set true to allow PassCore to  update the last password timestamp
    // The following options are for LDAP Provider (remove if you don't use this Provider)
    "LdapSearchBase": "ou=people,dc=example,dc=com",
    "LdapSecureSocketLayer": false, // Default for AD is true when using LDAPS 636
    "LdapStartTls": false, // Default for AD is true when using LDAP 389
    "LdapChangePasswordWithDelAdd": true,
    "LdapSearchFilter": "(sAMAccountName={Username})", // Another value: "(&(objectClass=person)(cn={Username}))"
    // General options (valid for both providers)
    "LdapHostnames": [ "" ], // Set your hostname(s)
    "LdapPort": 389, // Default for AD is 389, for LDAPS 636
    "LdapUsername": "", // Set the username or distinguish name (DN) to bind the LDAP server
    "LdapPassword": "", // Set the password for the username
    "DefaultDomain": "MyDomain.com" // Set your default AD domain here, or non "@" logins will not work! Use empty value to allow user to set the domain. This option is ONLY available with UPN.
  },
  "ClientSettings": {
    "ValidationRegex": {
      "EmailRegex": "^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\\.[a-zA-Z0-9-]+)*$",
      "UsernameRegex": "^[a-zA-Z0-9._-]{3,20}$"
    },
    "ShowPasswordMeter": true,
    "Recaptcha": {
      "SiteKey": "", // ReCAPTCHA public key: replace this! or leave empty if you don't need ReCAPTCHA
      "PrivateKey": "", // ReCAPTCHA private key: replace this! or leave empty if you don't need ReCAPTCHA
      "LanguageCode": "en"
    },
    "UseEmail": "false",
    "ApplicationTitle": "Ensure Technologies",
    "ChangePasswordTitle":  "XyLoc Password Reset Utility",
    "ChangePasswordForm": {
      "HelpText": "If you are having trouble with this tool, please contact IT Support",
      "UsernameLabel": "Username",
      "UsernameHelpblock": "Your organization's email address",
      "UsernameDefaultDomainHelperBlock": "Your organization's username",
      "CurrentPasswordLabel": "Current Password",
      "CurrentPasswordHelpblock": "Enter your current password",
      "NewPasswordLabel": "New Password",
      "NewPasswordHelpblock": "Enter a <a href='https://support.microsoft.com/en-us/help/4026406/microsoft-account-create-a-strong-password' target='_blank'>strong password</a>. You can use <a href='https://xkpasswd.net/s/' target='_blank'>this tool</a> to help you create one; use the XKCD (random sep, pad digit), or NTLM, options.",
      "NewPasswordVerifyLabel": "Re-enter New Password",
      "NewPasswordVerifyHelpblock": "Enter your new password again",
      "ChangePasswordButtonLabel": "Change Password"
    },
    "ErrorsPasswordForm": {
      "FieldRequired": "This field is required",
      "UsernamePattern": "Please enter a valid username",
      "UsernameEmailPattern": "Please enter a valid email address",
      "PasswordMatch": "Passwords do not match"
    },
    "Alerts": {
      "SuccessAlertTitle": "You have changed your password successfully.",
      "SuccessAlertBody": "Please note it may take a few hours for your new password to reach all domain controllers.",
      "ErrorPasswordChangeNotAllowed": "You are not allowed to change your password. Please contact your system administrator.",
      "ErrorInvalidCredentials": "You need to provide the correct current password.",
      "ErrorInvalidDomain": "You have supplied an invalid domain to logon to.",
      "ErrorInvalidUser": "We could not find your user account.",
      "ErrorCaptcha": "Could not verify you are not a robot.",
      "ErrorFieldRequired": "Fulfill all the fields.",
      "ErrorFieldMismatch": "The passwords do not match.",
      "ErrorConnectionLdap": "Unhandled error connecting to the LDAP server."
    }
  }
}

 
**Describe the bug**
I have in a test environment a windows 10 pro and a server 16 domain controller set up. I had initially put IIS on the windows 10 box. With this configuration it seemed that passcore was successfully changing the passwords. I took IIS off of the windows 10 box and put it onto the domain controller for testing. Now I only receive "Password must meet complexity requirements" errors when trying to change my password. I've been through all of the "closed" threads but am unable to resolve the issue. I've verified that i can change the password with both power shell and the command line without issue. Also, i have completely removed all password complexity requirements, age, and history

**To Reproduce**
1. Launch passcore
2. enter valid username, correct password, and valid password to change to
3. receive error "Password must meet complexity requirements"

**Desktop (please complete the following information):**
 - OS: windows 10 pro and server 16
 - Browser Internet explorer, chrome
 - Passcore version:  363

**Additional context**
All the setting were copied from the windows 10 IIS deployment. I'm just using local host to serve passcore

**Comment by DirectXGuy** [comment]
UPDATE:
I set this on on another test domain in our lab with identical settings and it changes passwords just fine and without issue.... strange

**Comment by sps-k12** [comment]
I am having this same issue with a brand new server 2016 member server.  The workstation where I'm trying to change the password is using chrome.  I have verified that I can change the users password to something very simple using several other methods, but passcore always reports that the password doesn't meet complexity requirements.  Even when I use the https://xkpasswd.net/s/ tool to create a really obnoxious password, it still fails the complexity requirements.

I have changed the default entropy settings and minimum score.  Are there invalid values here that can be causing this?  What are acceptable values for these fields?
